Examples/Framework/ML/src/NeuralCalibrator.cpp

```diff
@@ -102,81 +104,89 @@ void ActsExamples::NeuralCalibrator::calibrate(
   input[iInput++] = idxSourceLink.geometryId().layer();
 
   const Acts::Surface& referenceSurface = trackState.referenceSurface();
+  auto trackParameters = trackState.parameters();
+
+  const auto& measurement = measurements[idxSourceLink.index()];
+
+  assert(measurement.contains(Acts::eBoundLoc0) &&
+         "Measurement does not contain the required bound loc0");
+  assert(measurement.contains(Acts::eBoundLoc1) &&
+         "Measurement does not contain the required bound loc1");
+
+  auto boundLoc0 = measurement.subspace().indexOf(Acts::eBoundLoc0);
+  auto boundLoc1 = measurement.subspace().indexOf(Acts::eBoundLoc1);
+
+  Acts::Vector2 localPosition{measurement.effectiveParameters()[boundLoc0],
+                              measurement.effectiveParameters()[boundLoc1]};
+  Acts::Vector2 localCovariance{
+      measurement.effectiveCovariance()(boundLoc0, boundLoc0),
+      measurement.effectiveCovariance()(boundLoc1, boundLoc1)};
```

Comment:
These types of access could in principle also be done below inside the `visit_measurement`, but for pure parameter access, this probably doesn't matter. Another consideration would be caching the effective objects and reusing them for the access.